### DISCUSSION TITLE
lua的debug.traceback似乎有性能问题，不适合在性能要求高的函数里调用

### DISCUSSION THREAD

**Discussion by ghost** [cloudwu/skynet#1821] [General]
我为了记录玩家金币变化的调用，在金币变化的时候调用了debug.traceback。导致玩家每次在商店购买的请求的平均处理时间在0.1s左右。但是把debug.traceback去掉。商店购买的请求平均处理时间就能到0.0002左右。差距非常大。

**Comment by cloudwu** [comment]
debug.* api 都不适合放在正常的流程中，顾名思义，这些是 debug 用的。作为一个动态语言，应该动态的使用它们。

**Reply by ghost** [reply]
云大，捕获错误是不是也不能使用xpcall和debug.traceback的组合，如果错误高频触发，也会很影响性能
